src/test/java/phenex/task/TaskListTest.java

```diff
@@ -0,0 +1,43 @@
+package phenex.task;
+
+import org.junit.jupiter.api.Test;
+import phenex.exception.PhenexException;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.fail;
+
+/**
+ * Unit tests for the TaskList class.
+ */
+public class TaskListTest {
+
+    /**
+     * Tests that a phenex exception is thrown for inputting an invalid index when marking task.
+     */
+    @Test
+    public void invalid_index_phenexExceptionThrown() {
+        TaskList taskList = new TaskList();
+        ToDo toDoTask = new ToDo("task Test");
+        taskList.addTask(toDoTask);
```

Comment:
This part of the code is quite clear and easy to follow. I appreciate the effort to maintain readability while performing multiple operations.